### PR TITLE
feat: add account consumption

### PR DESCRIPTION
**Pull request by jsfez**
fixes ARG-46

### CODE REVIEW

File: apps/config/src/index.js [hunk 1]
```diff
@@ -131,6 +131,12 @@ const config = convict({
       default: "development",
       env: "GITHUB_WEBHOOK_SECRET",
     },
+    marketplaceUrl: {
+      doc: "Github marketplace url",
```
Comment (gregberge) [inline apps/config/src/index.js:135]:
```suggestion
      doc: "GitHub Marketplace URL",
```

File: apps/app/src/pages/Repository/BuildDetail/Screenshots.js [hunk 2]
```diff
@@ -97,8 +97,8 @@ function ScreenshotDiffItem({
 export default function BuildDetailScreenshots({ build }) {
   const [showPassingScreenshots, setShowPassingScreenshots] =
     React.useState(false);
-  const { screenshotDiffs } = build;
-  screenshotDiffs.sort((itemA, itemB) =>
+
+  const screenshotDiffs = [...build.screenshotDiffs].sort((itemA, itemB) =>
```
Comment (gregberge) [inline apps/app/src/pages/Repository/BuildDetail/Screenshots.js:101]:
```suggestion
  const screenshotDiffs = Array.from(build.screenshotDiffs);
  screenshotDiffs.sort((itemA, itemB) =>
```

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,5 @@ apps/*/dist/
 node_modules/
 *.pem
 .env
-diff_tmp.png
\ No newline at end of file
+diff_tmp.png
+yarn.lock
```
Comment (gregberge) [inline .gitignore:7]: We do not use yarn any more in the project, so it is off topic in the `.gitignore`.

File: apps/app/src/components/Link.js [hunk 4]
```diff
@@ -9,3 +11,21 @@ export const FadeLink = styled.aBox`
     opacity: 0.75;
   }
 `;
+
+const InnerLink = styled.aBox`
+  transition: base;
+  text-decoration: none;
+  color: #3291ff;
```
Comment (gregberge) [inline apps/app/src/components/Link.js:18]: Theme ?

File: apps/app/src/pages/Owner/GeneralSettings.js [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+const FeatureIcon = ({ icon, ...props }) => (
+  <Box as={icon} mr={2} mb="-2px" {...props} />
+);
+
+const PlanName = styled.box`
+  font-size: 36;
+
+  &::first-letter {
+    text-transform: uppercase;
+  }
```
Comment (gregberge) [inline apps/app/src/pages/Owner/GeneralSettings.js:35]: Should not be required, the name of the plan is the name of the plan.

File: apps/app/src/pages/Owner/GeneralSettings.js [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+        mx={-2}
+      >
+        <Card m={2} flex={1}>
+          <CardHeader>
+            <CardTitle>Plan</CardTitle>
+          </CardHeader>
+          <CardBody>
+            {currentPurchase?.plan ? (
+              <Plan
+                {...currentPurchase?.plan}
```
Comment (gregberge) [inline apps/app/src/pages/Owner/GeneralSettings.js:118]:
```suggestion
                {...currentPurchase.plan}
```

File: apps/app/src/pages/Owner/GeneralSettings.js [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+
+        {currentPurchase?.plan ? (
+          <Card m={2} flex={1}>
+            <CardHeader>
+              <CardTitle>Usage</CardTitle>
+            </CardHeader>
+            <CardBody minHeight="220px">
+              <CardTitle mb={2}>Screenshots used</CardTitle>
+              <Text fontSize={20}>
+                {Number(owner.currentMonthUsedScreenshots).toLocaleString()}
```
Comment (gregberge) [inline apps/app/src/pages/Owner/GeneralSettings.js:164]:
```suggestion
                {owner.currentMonthUsedScreenshots.toLocaleString()}
```

File: apps/app/src/pages/Owner/GeneralSettings.js [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        flexDirection={{ xs: "column", md: "row" }}
+        alignItems="flex-start"
+        mx={-2}
+      >
+        <Card m={2} flex={1}>
+          <CardHeader>
+            <CardTitle>Plan</CardTitle>
+          </CardHeader>
+          <CardBody>
+            {currentPurchase?.plan ? (
```
Comment (gregberge) [inline apps/app/src/pages/Owner/GeneralSettings.js:116]: If I read correctly the GraphQL schema, a purchase always have plan, so `currentPurchase` is sufficient to test here. The `?.plan` is not required.

**Review by gregberge — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (jsfez) [reply .gitignore:7]: Well, sometimes I forget about that. It was a convenient attempt to prevent regression

**Pull request merged**